cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -220,11 +221,48 @@ func Run(s *options.CMServer) error {
 	panic("unreachable")
 }
 
+func getAvailableResources(clientBuilder controller.ControllerClientBuilder) (map[unversioned.GroupVersionResource]bool, error) {
```

Comment:
am not a fan of using `map[...]bool` for sets. `map[...]struct{}` is the canonical type for that.